cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -58,6 +58,11 @@ func enforceRequirements(flags *applyPlanFlags, dryRun bool, newK8sVersion strin
 		return nil, errors.Wrapf(err, "couldn't create a Kubernetes client from file %q", flags.kubeConfigPath)
 	}
 
+	// Check if the cluster is self-hosted
+	if upgrade.IsControlPlaneSelfHosted(client) {
+		return nil, errors.Errorf("[upgrade/apply] Cannot upgrade a Self-Hosted control plane")
```

Comment:
`"[upgrade/apply] Cannot upgrade a Self-Hosted control plane"`
->
`"cannot upgrade a self-hosted control plane"`

Reply:
I am not sure why you want to remove the `[upgrade/apply]` Please explain so that I understand this logic in the future. This code is run only from plan and apply in the upgrade process.

Reply:
we are just consistent about not having the [....] prefix in return errors.
it's also a go pattern to have errors simple, propagate them up and leave it to the caller to handle formatting.